### PR TITLE
Use named indices for n-rank arrays

### LINKED ISSUE #251: Use named indices for n-rank arrays
For n-rank arrays (n > 1) it may be useful to give the indices names to understand the code faster.

### PR DESCRIPTION
**Pull request by matt-frey**
Closes #251.

### CODE REVIEW

File: src/3d/inversion/inversion.f90 [hunk 1]
```diff
@@ -244,47 +245,47 @@ subroutine vor2vel
...
-            double precision, intent(in) :: svel(0:nz, 0:nx-1, 0:ny-1, 3) ! velocity in semi-spectral space
-            double precision             :: ds(0:nz, 0:nx-1, 0:ny-1) ! semi-spectral derivatives
+            double precision, intent(in) :: svel(0:nz, 0:nx-1, 0:ny-1, n_dim) ! velocity in semi-spectral space
+            double precision             :: ds(0:nz, 0:nx-1, 0:ny-1)          ! semi-spectral derivatives
 
             ! x component:
-            call diffx(svel(:, :, :, 1), ds)         ! u_x = du/dx in semi-spectral space
-            call fftxys2p(ds, velgradg(0:nz, :, :, 1)) ! u_x in physical space
+            call diffx(svel(:, :, :, I_X), ds)           ! u_x = du/dx in semi-spectral space
+            call fftxys2p(ds, velgradg(0:nz, :, :, I_X)) ! u_x in physical space
```
Comment (sjboeing) [inline src/3d/inversion/inversion.f90:253]: Should this be I_DUDX?

Comment (matt-frey) [reply src/3d/inversion/inversion.f90:253]: Indeed. Although I_X = I_DUDX = 1. I will fix. Thanks for spotting this.

Comment (sjboeing) [reply src/3d/inversion/inversion.f90:253]: (realise the value is the same)

Comment (matt-frey) [reply src/3d/inversion/inversion.f90]: I hope you are fine with the naming convention if a leading `I_`.

File: src/3d/stepper/rk4_utils.f90 [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module rk4_utils
+    use dimensions, only : n_dim, I_X, I_Y, I_Z
+    use parcel_container, only : I_B11, I_B12, I_B13, I_B22, I_B23
     use parcel_ellipsoid, only : get_B33
-    use fields, only : velgradg, tbuoyg, vortg
+    use fields, only : velgradg, tbuoyg, vortg, I_DUDX, I_DUDY_I_DVDY, I_DWDX, I_DWDY
```
Comment (sjboeing) [inline src/3d/stepper/rk4_utils.f90:5]: Comma missing?

Comment (matt-frey) [reply src/3d/stepper/rk4_utils.f90:5]: Yes.

Comment (matt-frey) [reply src/3d/stepper/rk4_utils.f90]: Fixed.

**Review by sjboeing — COMMENTED** [review]
This makes the code clearer indeed! I had a look at this and caught 2 small possible issues. Otherwise, probably best to see if results still reproduce.

File: src/3d/inversion/inversion.f90 (hunk 1 above)
Comment (sjboeing) [reply src/3d/inversion/inversion.f90]: Yes, it is a good notation.

**Pull request merged**